### PR TITLE
Switch from Eto to WinForms

### LINKED ISSUE #4: Semicolon key press not registering as such
I don't know whether this is a bug in Eto, or whether I am using it wrong, or what, but when I type `;` (in Normal mode, which is the only operative mode right now) I should get an error message stating that the command is not yet implemented. However, `;` is failing to translate to `Void.Core.KeyPress.Semicolon`, because when the [key event](http://www.nudoq.org/#!/Packages/Eto.Forms/Eto/KeyPressEventArgs) comes down, it is not matching [this logic](https://github.com/PolyglotSymposium/void/blob/15e081ad857de0078d32c118528443983f838a99/Void/Void/TypeTranslationExtensions.cs#L316).

### PR DESCRIPTION
**Pull request by Kazark**
This resolves #4 and #5 as well as several other pain-points for which issues have not been submitted (such as Intellisense not working in the C# code). It lays a good foundation for solving #3. Unfortunately it does not solve #2 yet, which I was hoping for. However, because it now seems clear to me that WinForms is better for our purposes than Eto, and because this branch has diverged a good bit, I thought I ought to submit a pull request.

This also contains many improvements not directly tied to WinForms that however seemed necessary to do as I went. If we still want to experiment with other GUI frameworks such as WPF that is fine but we will want to merge this first and then branch from here to try new ones. This is a good new basis for things.

### CODE REVIEW

File: Void/Void.ViewModel/VoidController.fs [hunk 1]
```diff
@@ -39,67 +43,86 @@ type ViewController
...
-        | Command.ViewTestBuffer -> 
-            _coreCtrl.handleCommand command |> x.handleCommand
+                handle event |> x.brokerCommand
         | Command.Noop -> ()
+        | _ ->
+            for handle in _commandHandlers do
+                handle command |> x.brokerCommand
+
+module Init =
+    let initializeVoid view =
```
Comment (mjgpy3) [inline Void/Void.ViewModel/VoidController.fs:109]: I really like the explicit wiring here.

Comment (Kazark) [inline Void/Void.ViewModel/VoidController.fs:109]: Inspired by a comment you made that it didn't make sense to have it in the `VoidController`. :)

**Comment by mjgpy3** [comment]
If you're alright with it, I'm pretty much ready to merge this. There are a lot of changes (and I definitely don't comprehend them all) but I think it looks good :+1:

**Comment by Kazark** [comment]
Merge away!

**Pull request merged**